### ISSUE TITLE
No proper way to create customizable ModelResource

### ISSUE THREAD

**Issue by andreynovikov** [django-import-export/django-import-export#1034]
I needed to customize my `ModelResource` behavior for each import. The most straightforward way is to add extra field to `ImportForm`. But there is no way to access form data from `ModelResource`.

The first issue is that `get_import_resource_kwargs` knows nothing about import forms. Thanks it knows about `request`. So the first ugly hack was to recreate and re-validate my custom import form inside this method.

But the next issue is even worse: custom field value from `ImportForm` should be passed through `ConfirmImportForm` but the solution from documentation can not work. See:
https://github.com/django-import-export/django-import-export/blob/3cf5e3f9796a5caf7c5ea3928119af7ce4706c0d/import_export/admin.py#L322
Fortunately there is currently a typo in this line, it obviously should be:

```python
initial = self.get_form_kwargs(form=confirm_form, **initial)
```

This typo makes possible second ugly hack to access `ImportForm` data to initialize `ConfirmImportForm`. But when you will fix it there would be no way to do that.

I suggest this part of import workflow should be redesigned.

**Comment by mgrdcm** [comment]
@andreynovikov I think this might be solved now in [2.0.2](https://github.com/django-import-export/django-import-export/blob/master/docs/changelog.rst#210-2020-05-02) by #994, #1108.  Can you confirm?

**Comment by imomaliev** [comment]
Should be fixed by #1185

**Comment by andrewgy8** [comment]
This seems to be solved. I will close it for now.
